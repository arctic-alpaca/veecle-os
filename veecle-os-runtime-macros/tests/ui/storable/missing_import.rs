mod inner {
    pub struct Inner;
}

#[derive(Debug, PartialEq, veecle_os_runtime_macros::Storable)]
#[storable(data_type = "Vec<Inner>")]
pub struct Sensor;

fn value() -> u8 {
    1
}
#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(
    _sensor_reader: veecle_os_runtime::Reader<'_, Sensor>,
    _sensor_writer: veecle_os_runtime::Writer<'_, Sensor>,
) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

fn main() {
    let _ = veecle_os_runtime::execute! {
        store: [Sensor],
        actors: [MacroTestActor],
    };
}
