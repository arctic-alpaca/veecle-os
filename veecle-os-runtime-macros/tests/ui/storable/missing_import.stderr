error[E0412]: cannot find type `Inner` in this scope
 --> tests/ui/storable/missing_import.rs:6:24
  |
6 | #[storable(data_type = "Vec<Inner>")]
  |                        ^^^^^^^^^^^^ not found in this scope
  |
help: consider importing this struct
  |
1 + use crate::inner::Inner;
  |
